crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -85,6 +85,24 @@ pub(super) fn place_comment<'a>(
             handle_leading_class_with_decorators_comment(comment, class_def)
         }
         AnyNodeRef::StmtImportFrom(import_from) => handle_import_from_comment(comment, import_from),
+        AnyNodeRef::StmtWhile(stmt_while) => {
+            handle_terminal_comment(comment, &TerminalExpression::from(stmt_while), locator)
+        }
+        AnyNodeRef::StmtIf(stmt_if) => {
+            handle_terminal_comment(comment, &TerminalExpression::from(stmt_if), locator)
+        }
+        AnyNodeRef::StmtFor(stmt_for) => {
+            handle_terminal_comment(comment, &TerminalExpression::from(stmt_for), locator)
+        }
+        AnyNodeRef::StmtAsyncFor(stmt_for) => {
+            handle_terminal_comment(comment, &TerminalExpression::from(stmt_for), locator)
+        }
+        AnyNodeRef::StmtWith(stmt_with) => {
+            handle_terminal_comment(comment, &TerminalExpression::from(stmt_with), locator)
+        }
+        AnyNodeRef::StmtAsyncWith(stmt_with) => {
+            handle_terminal_comment(comment, &TerminalExpression::from(stmt_with), locator)
+        }
```

Comment:
You don't need `TerminalExpression`: 405740a8ee6da3b37e927d612e501b4caa114b6a

Reply:
I kind of like it (besides the name)

Reply:
in that case you can remove the `From` impls and instead pass the values directly to the struct

Reply:
FWIW I kind of like this pattern with the struct, though agree it may be better to avoid the `from`.